codegen/winterfell/src/air/boundary_constraints.rs

```diff
@@ -167,7 +167,11 @@ impl Codegen for Expression {
                 )
             }
             Self::Exp(lhs, rhs) => {
-                format!("({}).exp({})", lhs.to_string(ir, is_aux_constraint), rhs)
+                if let Self::Const(rhs) = **rhs {
+                    format!("({}).exp({})", lhs.to_string(ir, is_aux_constraint), rhs)
+                } else {
+                    todo!()
+                }
```

Comment:
I requested implementing the IR change now, but leaving this `todo` is fine, as this whole section changes with my current PR #126, so we can update it during rebase